### PR TITLE
Add `settings` and `parser` options

### LINKED ISSUE #138: No way to pass `settings`?
The eslint-import checker is nice but it needs to be told about my webpack config. However, I don't see how to provide the `settings` key to eslint.

### PR DESCRIPTION
**Pull request by wmertens**
Closes #138.

### CODE REVIEW

**Comment by wmertens** [comment]
Weird thing though, when I use this to set the `import/resolver`, it makes it into the config that is generated for ESLint, but the import plugin only gets the rule set by the `plugins` extend.

generated config:
```
{ rules: 
   { strict: 0,
     quotes: 0,
     eqeqeq: [ 2, 'allow-null' ],
     'no-eq-null': 0,
     'operator-linebreak': [ 2, 'after', [Object] ],
     'brace-style': [ 2, '1tbs', [Object] ],
     'babel/arrow-parens': 0,
     'comma-dangle': [ 2, 'always-multiline' ],
     'react/jsx-closing-bracket-location': [ 2, 'tag-aligned' ],
     'react/prop-types': 1,
     'react/prefer-stateless-function': 1,
     'prefer-template': 0,
     semi: [ 2, 'never' ],
     'semi-spacing': [ 2, [Object] ] },
  settings: { 'import/resolver': { webpack: [Object] } },
  globals: [ '$', 'info', 'dbg', 'views', '_dict', 'dict', 'lang' ],
  envs: [ 'browser', 'mocha', 'webpack' ],
  plugins: [ 'react', 'babel' ],
  extends: [ 'xo-react' ],
  useEslintrc: false,
  cache: true,
  cacheLocation: '/Users/wmertens/.xo-cache/',
  baseConfig: 
   { extends: 
      [ 'xo/esnext',
        '/Users/wmertens/.../src/node_modules/xo/config/plugins.js',
        '/Users/wmertens/.../src/node_modules/eslint-config-xo-react/index.js' ] },
  esnext: true,
  semicolon: false,
  cwd: '/Users/wmertens/.../src',
  init: false,
  compact: false,
  stdin: false,
  fix: false,
  open: false,
  ignores: 
   [ '**/node_modules/**',
     '**/bower_components/**',
     'coverage/**',
     '{tmp,temp}/**',
     '**/*.min.js',
     '**/bundle.js',
     'fixture{-*,}.{js,jsx}',
     'fixture{s,}/**',
     '{test,tests,spec,__tests__}/fixture{s,}/**',
     'vendor/**',
     'dist/**' ] }
```

settings that the import plugin receives:
```
{ 'import/extensions': [ '.js' ] }
```

Does eslint not properly merge the settings?

**Comment by wmertens** [comment]
Meep?

File: test/options-manager.js [hunk 1]
```diff
@@ -73,6 +75,12 @@ test('buildConfig: rules', t => {
 	t.deepEqual(config.rules, rules);
 });
 
+test('buildConfig: settings', t => {
+	const settings = {'import/resolver': 'webpack'};
+	const config = manager.buildConfig({settings: settings});
```
Comment (sindresorhus) [inline test/options-manager.js]:
```js
const config = manager.buildConfig({settings});
```

**Comment by sindresorhus** [comment]
> Weird thing though, when I use this to set the import/resolver, it makes it into the config that is generated for ESLint, but the import plugin only gets the rule set by the plugins extend.

// @jfmengels in case you have any idea what could cause this.

**Comment by jfmengels** [comment]
Sorry, don't have a clue :confused:

**Comment by wmertens** [comment]
Done

**Comment by sindresorhus** [comment]
Ok, figured it out. It's because we pass these option to the [ESLint Node.js API](http://eslint.org/docs/developer-guide/nodejs-api), not as normal config (since it's generated). The ESLint Node.js API doesn't yet have a `settings` option. @wmertens Can you open an issue on ESLint about adding that?

**Comment by wmertens** [comment]
@sindresorhus it turns out you're supposed to pass the config via `baseconfig`…

**Comment by sindresorhus** [comment]
@wmertens Ah ok. Makes sense. Can you update to use `baseConfig` instead? (See `extends` for example).

**Comment by sindresorhus** [comment]
No pressure on this, but I plan on doing a new release in a few days, so this needs to be updated if you want it included ;)

**Comment by wmertens** [comment]
Noted, will try ☺

**Comment by sindresorhus** [comment]
Last chance. Tomorrow is release day ;)

**Comment by wmertens** [comment]
@sindresorhus Alright, this should do the trick…

**Comment by sindresorhus** [comment]
Awesome. Glad we could land this before the release. Thank you @wmertens :)

**Pull request merged**